WCF/Shared/Implementation/WcfOperationContext.cs

```diff
@@ -115,7 +129,7 @@ private static IOperationContext GetContext()
             {
                 if ( owner != null )
                 {
-                    context = new WcfOperationContext(owner);
+                    context = new WcfOperationContext(owner, HttpContext.Current);
```

Comment:
The only recommendation I have is to detect if an app running in IIS somehow and do not attempt to get the http context every time. It will be better from performance perspective